wrappers/services.go

```diff
@@ -471,13 +471,48 @@ func tryFileUpdate(path string, desiredContent []byte) (old *osutil.MemoryFileSt
 	}
 }
 
-type SnapServiceOptions struct {
+type ServiceQuotaMap map[*snap.AppInfo]*quota.Group
+
+// MakeServiceQuotaMap builds a map of services and their quota groups. The goal is
+// to make sure that if any quota group applies to a snap service, then we can easily
+// look it up in this based on it's snap.AppInfo. We take the snap instance name, the services
+// want to include in the map (should only be services of the snap name provided) and the quota
+// group of the snap.
+func MakeServiceQuotaMap(snapName string, svcs []*snap.AppInfo, grp *quota.Group) ServiceQuotaMap {
+	if len(svcs) == 0 {
+		return nil
+	}
+
+	svcQuotaMap := make(ServiceQuotaMap, len(svcs))
+	for _, svc := range svcs {
+		// set nil grps for all services if parent is nil
+		if grp == nil {
+			svcQuotaMap[svc] = nil
```

Comment:
nil is the default value one gets for missing keys in a pointer values map, so

Reply:
Yes true, but the point is also having the keys values. We use the map as a list of services as well. It serves both purposes of having a group set, but also provide a list. So we must fill in the keys with nil values as well.

Reply:
Yes, but that is really just nice but the conceptual confusion is more of an issue. wrappers does have the info and can find out the list of services, that's also why I made the point that is not clear that this helper needs to fill the default snap group. It could be the caller that does that, which would make the caller happy with nil for the map. Basically I think Services should become more like ServiceGroup and be filled only if the group is not the snap group. And wrappers should deal with that.

Reply:
I changed the approach to only send groups that has quota groups down to wrappers and then cross-reference with that list in wrappers. Thank you for all the input!